File: PhpBeginner/2020/EarlyPeriod/comp.php
<?php   // comp.php
//  http://dev2.m-fr.net/アカウント名/comp.php

var_dump( 1 === 1 );
var_dump( 1 === 2 );
var_dump( 1 == '1' );
var_dump( 1 === '1' );
var_dump( 1 < 2 );
echo "<br>\n";
//
var_dump( (bool)'abc' );
var_dump( (bool)'' );
var_dump( (bool)1 );
var_dump( (bool)0 );
var_dump( (bool)-1 ); // true
var_dump( (bool)[1,2,3] );
var_dump( (bool)[] );
var_dump( (bool)null );
var_dump( (bool)'1' );
var_dump( (bool)'0' ); // XXX false

File: C/2020/56限/EarlyPeriod/comp.c
//  comp.c
//  cl  comp.c
#include <stdio.h>

int main() {
    //
    if (1) {
        printf("1 is true\n");
    }
    if (0) {
        printf("0 is true\n");
    }
    if (-1) {
        printf("-1 is true\n");
    }
    //
    int i = 0;
    printf("1 == 1 is %d \n", 1 == 1);
    printf("1 == 2 is %d \n", 1 == 2);
    printf("1 < 2 is %d \n", 1 < 2);
    printf("1 > 2 is %d \n", 1 > 2);
    printf("i = 20 is %d \n", i = 20);

    return 0;
}
